Libraries/LibCore/File.cpp

```diff
@@ -132,6 +134,30 @@ String File::real_path_for(const String& filename)
     return real_path;
 }
 
+bool File::ensure_parent_directories(const String& path)
+{
+    ASSERT(path[0] == '/');
+
+    char* parent_buffer = strdup(path.characters());
+    const char* parent = dirname(parent_buffer);
+
+    int rc = 0;
+    while (true) {
+        int rc = mkdir(parent, 0755);
```

Comment:
You're shadowing the `rc` from outer scope; so the outer scope `rc` will never get set to anything but 0.